lib/charms/layer/openstack.py

```diff
@@ -95,13 +96,28 @@ def _save_creds(creds_data):
     else:
         attrs = creds_data
         endpoint = attrs['auth-url']
+
+    if 'ca-certificates' in creds_data:
+        # see K8s commit e3c8a0ceb66816433b095c4d734663e1b1e0e4ea
+        # K8s in-tree cloud provider code is not flexible enough
+        # to accept multiple certs that could be provided by Juju
+        # so we can grab the first one only and hope it is the
+        # right one
+        ca_certificates = creds_data.get('ca-certificates')
+        ca_cert = ca_certificates[0] if ca_certificates else None
+    elif 'endpoint-tls-ca' in creds_data:
```

Comment:
I'd prefer to continue the pattern of having the name and format of the config values match that returned by `credentials-get` so that we don't have to maintain this special-case logic to distinguish between config-provided or Juju provided data.

Reply:
Do you mean changing endpoint-tls-ca to ca-certificates?